ownCloud/Theming/NSObject+ThemeApplication.swift

```diff
@@ -271,3 +291,48 @@ extension UITableViewController : ThemeableSectionHeader, ThemeableSectionFooter
 		}
 	}
 }
+
+extension UISearchBar {
+
+	func getTextField() -> UITextField? {
+			return value(forKey: "searchField") as? UITextField
+
+	}
+	func setPlaceholder(textColor: UIColor) {
+		getTextField()?.setPlaceholder(textColor: textColor)
+
+	}
+}
+
+private extension UITextField {
+
+    private class Label: UILabel {
+        private var _textColor = UIColor.lightGray
+        override var textColor: UIColor! {
+            set { super.textColor = _textColor }
+            get { return _textColor }
+        }
+
+        init(label: UILabel, textColor: UIColor = .lightGray) {
+            _textColor = textColor
+            super.init(frame: label.frame)
+            self.text = label.text
+            self.font = label.font
+        }
+
+        required init?(coder: NSCoder) { super.init(coder: coder) }
+    }
+
+    var placeholderLabel: UILabel? {
+		return value(forKey: "placeholderLabel") as? UILabel
+
+	}
+
+    func setPlaceholder(textColor: UIColor) {
+        guard let placeholderLabel = placeholderLabel else { return }
+
+        let label = Label(label: placeholderLabel, textColor: textColor)
+        placeholderLabel.removeFromSuperview() // To remove existing label. Otherwise it will overwrite it if called multiple times.
+        setValue(label, forKey: "placeholderLabel")
```

Comment:
IMO this should be avoided as it's likely to break (and cause a crash) in the future. The color should be changeable with public API by using `attributedPlaceholder` and passing an `NSAttributedString` with the desired foreground color and font.

Reply:
Yes, it is possible with a workaround, because of an iOS 13 bug, the attributed string will only be rendered in `viewDidLayoutSubview` and not int `viewDidLoad`. Removed unneeded code.